src/RoslynPad.Roslyn/WorkspaceExtensions.cs

```diff
@@ -0,0 +1,20 @@
+using Microsoft.CodeAnalysis;
+
+namespace RoslynPad.Roslyn
+{
+    /// <summary>
+    /// Provide a way for users to turn on and off analyzing workspace for compiler diagnostics
+    /// </summary>
+    public static class WorkspaceExtensions
+    {
+        public static void EnableDiagnostics(this Workspace workspace, DiagnosticOptions options)
+        {
+            DiagnosticProvider.Enable(workspace, (DiagnosticProvider.Options)options);
```

Comment:
Instead of a casting the options use a switch. It's more resilient to changes, esp since this is internal code.